From 5758cb14681541dfab829d1ae204c861c05072a7 Mon Sep 17 00:00:00 2001
From: Muna Sinada <quic_msinada@quicinc.com>
Date: Mon, 4 Oct 2021 11:05:51 -0700
Subject: [PATCH] tests: Initial EHT testing

Add initial tests for EHT.

Tests include the following:

 - eht80_to_24g_eht - EHT with 80 MHz channel width reconfigured to
		      2.4 GHz EHT
 - eht_40 - EHT and 40 MHz channel
 - eht_use_sta_nsts - EHT with 80 MHz channel width and use_sta_nsts=1
 - eht_open - EHT AP with open mode configuration
 - eht80_invalid2 - EHT with invalid 80 MHz channel configuration (seg0)
 - eht80b - EHT with 80 MHz channel width (HT40- channel 40)
 - eht80d - EHT with 80 MHz channel width (HT40- channel 48)
 - eht_on_24ghz - Subset of EHT features on 2.4 GHz
 - eht_params - EHT AP parameters
 - eht_wifi_generation_24
 - eht_wifi_generation
 - eht80 - EHT with 80 MHz channel width
 - eht_20 - EHT and 20 MHz channel
 - eht_disabled_on_sta - EHT AP and EHT STA disabled on STA
 - eht80c - EHT with 80 MHz channel width (HT40+ channel 44)
 - eht80_params - EHT with 80 MHz channel width and number of optional
		  features enabled
 - eht80_csa - EHT with 80 MHz channel width and CSA
 - eht160 - EHT with 160 MHz channel width [long]
 - eht80_invalid - EHT with invalid 80 MHz channel configuration
		   (seg1: 80+80 not supported)

Signed-off-by: Muna Sinada <quic_msinada@quicinc.com>
---
 tests/hwsim/example-hostapd.config        |   1 +
 tests/hwsim/example-wpa_supplicant.config |   2 +
 tests/hwsim/test_eht.py                   | 691 ++++++++++++++++++++++++++++++
 tests/hwsim/wpasupplicant.py              |   2 +-
 4 files changed, 695 insertions(+), 1 deletion(-)
 create mode 100644 tests/hwsim/test_eht.py

--- a/tests/hwsim/example-hostapd.config
+++ b/tests/hwsim/example-hostapd.config
@@ -53,6 +53,7 @@ CONFIG_LIBNL3_ROUTE=y
 CONFIG_IEEE80211R=y
 CONFIG_IEEE80211AC=y
 CONFIG_IEEE80211AX=y
+CONFIG_IEEE80211BE=y
 
 CONFIG_OCV=y
 
--- a/tests/hwsim/example-wpa_supplicant.config
+++ b/tests/hwsim/example-wpa_supplicant.config
@@ -63,6 +63,7 @@ CONFIG_CTRL_IFACE_DBUS_INTRO=y
 CONFIG_IEEE80211R=y
 CONFIG_IEEE80211AC=y
 CONFIG_IEEE80211AX=y
+CONFIG_IEEE80211BE=y
 
 CONFIG_OCV=y
 
@@ -106,6 +107,7 @@ CONFIG_TLSV12=y
 CONFIG_HT_OVERRIDES=y
 CONFIG_VHT_OVERRIDES=y
 CONFIG_HE_OVERRIDES=y
+CONFIG_EHT_OVERRIDES=y
 
 CONFIG_DEBUG_LINUX_TRACING=y
 
--- a/tests/hwsim/wpasupplicant.py
+++ b/tests/hwsim/wpasupplicant.py
@@ -1094,7 +1094,7 @@ class WpaSupplicant:
                       "wep_tx_keyidx", "scan_freq", "freq_list", "eap",
                       "eapol_flags", "fragment_size", "scan_ssid", "auth_alg",
                       "wpa_ptk_rekey", "disable_ht", "disable_vht", "bssid",
-                      "disable_he",
+                      "disable_he", "disable_eht",
                       "disable_max_amsdu", "ampdu_factor", "ampdu_density",
                       "disable_ht40", "disable_sgi", "disable_ldpc",
                       "ht40_intolerant", "update_identifier", "mac_addr",
